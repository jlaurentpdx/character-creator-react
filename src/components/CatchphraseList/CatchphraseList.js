export default function CatchphraseList({ headCount, midCount, pantsCount, catchphraseList }) {
  return (
    <div className="stats">
      <p className="display-stats hidden">
        You have changed the head {headCount} times, the body {midCount} times, and the pants{' '}
        {pantsCount} times. And nobody can forget your character&apos;s classic catchphrases:
      </p>
      {catchphraseList.map((item) => (
        <p key={item} className="catchphrase">
          {item}
        </p>
      ))}
    </div>
  );
}
