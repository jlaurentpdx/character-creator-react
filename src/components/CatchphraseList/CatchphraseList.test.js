import { render, screen } from '@testing-library/react';
import CatchphraseList from './CatchphraseList';

test('should render stats and a list of catchphrases', () => {
  const { container } = render(
    <CatchphraseList
      headCount={1}
      midCount={2}
      pantsCount={0}
      catchphraseList={['oh gee', 'testing is a hoot']}
    />
  );
  screen.getByText(
    `You have changed the head 1 times, the body 2 times, and the pants 0 times. And nobody can forget your character's classic catchphrases:`
  );
  screen.getByText('oh gee');
  screen.getByText('testing is a hoot');

  expect(container).toMatchSnapshot();
});
