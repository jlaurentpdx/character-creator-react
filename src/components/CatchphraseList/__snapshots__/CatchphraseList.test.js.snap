// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stats and a list of catchphrases 1`] = `
<div>
  <div
    class="stats"
  >
    <p
      class="display-stats hidden"
    >
      You have changed the head 
      1
       times, the body 
      2
       times, and the pants
       
      0
       times. And nobody can forget your character's classic catchphrases:
    </p>
    <p
      class="catchphrase"
    >
      oh gee
    </p>
    <p
      class="catchphrase"
    >
      testing is a hoot
    </p>
  </div>
</div>
`;
